x/pool-incentives/keeper/grpc_query.go

```diff
@@ -152,6 +154,48 @@ func (q Querier) IncentivizedPools(ctx context.Context, _ *types.QueryIncentiviz
 					GaugeId:          record.GaugeId,
 				}
 
+				incentivizedPools = append(incentivizedPools, incentivizedPool)
+				incentivizedPoolIDs[poolId] = struct{}{}
+			}
+		}
+	}
+
+	// Only run the following if the above loop determined there were incentivized pools.
+	if len(incentivizedPoolIDs) > 0 {
+		// Retrieve the migration records between balancer pools and concentrated liquidity pools.
+		// This comes from the superfluid keeper, since superfluid is the only pool incentives connected
+		// module that has access to the gamm modules store.
+		migrationRecords, err := q.superfluidKeeper.GetAllMigrationInfo(sdkCtx)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+
+		// Iterate over all migration records.
+		for _, record := range migrationRecords.BalancerToConcentratedPoolLinks {
+			// If the cl pool is not in the list of incentivized pools, skip it.
+			_, incentivized := incentivizedPoolIDs[record.ClPoolId]
+			if !incentivized {
+				continue
+			}
+
+			// Determine the duration of the incentivized pool.
+			duration := lockableDurations[len(lockableDurations)-1]
```

Comment:
Let's use `GetLongestLockableDuration` instead please. It does a length check, making this safer

Reply:
https://github.com/osmosis-labs/osmosis/pull/5589/commits/1ebe6dc5eed91e8cd979beab886f9f78efe92d1c